craft_parts/errors.py

```diff
@@ -482,15 +502,51 @@ class PluginBuildError(PartsError):
     :param plugin_name: The name of the plugin being processed.
     """
 
-    def __init__(self, *, part_name: str, plugin_name: str) -> None:
+    def __init__(
+        self, *, part_name: str, plugin_name: str, stderr: bytes | None = None
+    ) -> None:
         self.part_name = part_name
         self.plugin_name = plugin_name
+        self.stderr = stderr
         brief = f"Failed to run the build script for part {part_name!r}."
+
         resolution = f"Check the build output and verify the project can work with the {plugin_name!r} plugin."
         super().__init__(
             brief=brief, resolution=resolution, doc_slug="/reference/plugins.html"
         )
 
+    @property
+    @override
+    def brief(self) -> str:
+        """A brief summary of the error.
+
+        Discards all trace lines that come before the last-executed script line
+        """
+        brief = f"Failed to run the build script for part {self.part_name!r}."
+
+        if self.stderr is None:
+            return brief
+
+        stderr = self.stderr.decode("utf-8", errors="replace")
+        brief += "\nCaptured standard error:"
+
+        stderr_lines = stderr.split("\n")
+        # Find the final command captured in the logs
+        last_command = None
+        for idx, line in enumerate(reversed(stderr_lines)):
+            if line.startswith("+"):
+                last_command = len(stderr_lines) - idx - 1
+                break
+        else:
+            # Fallback to printing the whole log
+            last_command = 0
+
+        for line in stderr_lines[last_command:]:
+            if line:
+                brief += f"\n:: {line}"
```

Comment:
It is generally not recommended in Python to build a string by using `+` operator due to its inefficient implementation .
Logs can grow pretty big and you may consider using a better method to build a final string (like using `StringIO`).